extensions/control-plane/api/management-api/secret-api/src/main/java/org/eclipse/edc/connector/api/management/secret/SecretApiController.java

```diff
@@ -0,0 +1,136 @@
+
+import static jakarta.json.stream.JsonCollectors.toJsonArray;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON;
+import static java.util.Optional.of;
+import static org.eclipse.edc.spi.query.QuerySpec.EDC_QUERY_SPEC_TYPE;
+import static org.eclipse.edc.spi.types.domain.secret.Secret.EDC_SECRET_TYPE;
+import static org.eclipse.edc.web.spi.exception.ServiceResultHandler.exceptionMapper;
+
+@Consumes(APPLICATION_JSON)
+@Produces(APPLICATION_JSON)
+@Path("/v3/secrets")
+public class SecretApiController implements SecretApi {
+    private final TypeTransformerRegistry transformerRegistry;
+    private final SecretService service;
+    private final Monitor monitor;
+    private final JsonObjectValidatorRegistry validator;
+
+    public SecretApiController(SecretService service, TypeTransformerRegistry transformerRegistry,
+                               Monitor monitor, JsonObjectValidatorRegistry validator) {
+        this.transformerRegistry = transformerRegistry;
+        this.service = service;
+        this.monitor = monitor;
+        this.validator = validator;
+    }
+
+    @POST
+    @Override
+    public JsonObject createSecret(JsonObject secretJson) {
+        validator.validate(EDC_SECRET_TYPE, secretJson).orElseThrow(ValidationFailureException::new);
+
+        var secret = transformerRegistry.transform(secretJson, Secret.class)
+                .orElseThrow(InvalidRequestException::new);
+
+        var idResponse = service.create(secret)
+                .map(a -> IdResponse.Builder.newInstance()
+                        .id(a.getId())
+                        .createdAt(a.getCreatedAt())
+                        .build())
+                .orElseThrow(exceptionMapper(Secret.class, secret.getId()));
+
+        return transformerRegistry.transform(idResponse, JsonObject.class)
+                .orElseThrow(f -> new EdcException(f.getFailureDetail()));
+    }
+
+    @POST
+    @Path("/request")
+    @Override
+    public JsonArray requestSecrets(JsonObject querySpecJson) {
+        QuerySpec querySpec;
+        if (querySpecJson == null) {
+            querySpec = QuerySpec.Builder.newInstance().build();
+        } else {
+            validator.validate(EDC_QUERY_SPEC_TYPE, querySpecJson).orElseThrow(ValidationFailureException::new);
+
+            querySpec = transformerRegistry.transform(querySpecJson, QuerySpec.class)
+                    .orElseThrow(InvalidRequestException::new);
+        }
+
+        return service.search(querySpec).orElseThrow(exceptionMapper(QuerySpec.class, null)).stream()
+                .map(it -> transformerRegistry.transform(it, JsonObject.class))
+                .peek(r -> r.onFailure(f -> monitor.warning(f.getFailureDetail())))
+                .filter(Result::succeeded)
+                .map(Result::getContent)
+                .collect(toJsonArray());
+    }
+
+    @GET
+    @Path("{id}")
+    @Override
+    public JsonObject getSecret(@PathParam("id") String id) {
```

Comment:
putting the ID in the path will cause it to be transmitted unencrypted. I would transmit the secret ID in the body. same for the `delete` api.

Reply:
as long as TLS is used, the URL will be encrypted, to me it's ok to keep it in the url.
also because the secret key is not really a "sensitive" value

Reply:
yeah, true, it's only the host part that goes unencrypted. I'm OK with leaving it in the path since that makes things a bit easier.